src/avro_fingerprint.erl

```diff
@@ -0,0 +1,344 @@
+%% coding: latin-1
+%%%-----------------------------------------------------------------------------
+%%%
+%%% Copyright (c) 2018 Klarna AB
+%%%
+%%% This file is provided to you under the Apache License,
+%%% Version 2.0 (the "License"); you may not use this file
+%%% except in compliance with the License.  You may obtain
+%%% a copy of the License at
+%%%
+%%%   http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%% Unless required by applicable law or agreed to in writing,
+%%% software distributed under the License is distributed on an
+%%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%%% KIND, either express or implied.  See the License for the
+%%% specific language governing permissions and limitations
+%%% under the License.
+%%%
+%%% @author Jake Morrison <jake@cogini.com>
+%%%
+%%% @doc
+%%% Implement the Avro CRC 64 fingerprint algorithm.
+%%%
+%%% The algorithm is defined with this Java code:
+%%%
+%%%  ```
+%%%  static long fingerprint64(byte[] buf) {
+%%%    if (FP_TABLE == null) initFPTable();
+%%%    long fp = EMPTY;
+%%%    for (int i = 0; i < buf.length; i++)
+%%%      fp = (fp >>> 8) ^ FP_TABLE[(int)(fp ^ buf[i]) & 0xff];
+%%%    return fp;
+%%%  }
+%%%
+%%%  static long EMPTY = 0xc15d213aa4d7a795L;
+%%%  static long[] FP_TABLE = null;
+%%%
+%%%  static void initFPTable() {
```

Comment:
We can have this function implemented in Erlang too to avoid hardcoding.
i.e. Generate a length=256 integer array `List`, then Table = `list_to_tuple(List)`.
Use `element(Byte, Table)` to lookup, it should have about the same performance as function clause pattern matching.
The overhead of this init function is OK because I don't think fingerprint calculation would be intensive in any use case.

Reply:
Seems better to just do it once, rather than at run time. And it is certainly compatible with the Java side.

Reply:
ok.